### PR TITLE
Fixing broken returns

### PR DESCRIPTION
**Pull request by john-science**
## Description

A ton of our functions and methods have bad return statements, I am attempting to fix those here.

**NOTE**: This is a WIP, as I expect this to drop our code coverage a lot. And I need to see that code coverage difference and fix it in tests.

---

## Checklist

- [X] This PR has only one purpose or idea.
- [x] Tests have been added/updated to verify that the new/changed code works.

- [X] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [X] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [X] The [release notes](https://terrapower.github.io/armi/release/index.html) are up-to-date with any bug fixes or new features.
- [X] The documentation is still up-to-date in the `doc` folder.
- [X] The dependencies are still up-to-date in `setup.py`.

### CODE REVIEW

File: armi/utils/codeTiming.py [hunk 1]
```diff
@@ -404,13 +404,12 @@ def start(self):
...
 
         if self._frozen:
-            return
-
-        if self.isActive:
-            self.over_start += (
-                1  # call was made on an active timer, we're now over-started
-            )
+            return cur_time
+        elif self.isActive:
```
Comment (opotowsky) [inline armi/utils/codeTiming.py:408]: Should this just be `else:`?

File: armi/utils/codeTiming.py [hunk 2]
```diff
@@ -420,9 +419,10 @@ def stop(self):
...
 
         if self._frozen:
-            return
+            return cur_time
 
-        if self.over_start:  # can't end the timer as it's over-started
+        if self.over_start:
+            # can't end the timer as it's over-started
             self.over_start -= 1
         elif self.isActive:
```
Comment (opotowsky) [inline armi/utils/codeTiming.py:427]: Should this just be `else:`? Or a separate if statement?

**Review by opotowsky — APPROVED** [review]
I only have a couple of questions, nothing blocking!

File: armi/utils/codeTiming.py (hunk 1 above)
Comment (john-science) [reply armi/utils/codeTiming.py:408]: That's a good call. I'll do that!

File: armi/utils/codeTiming.py (hunk 2 above)
Comment (john-science) [reply armi/utils/codeTiming.py:427]: Oh, nope. That failed. 

It appears there are several more options than active and frozen.z

Now we know.

File: armi/utils/codeTiming.py (hunk 1 above)
Comment (john-science) [reply armi/utils/codeTiming.py:408]: Oh, nope. That failed. 

It appears there are several more options than active and frozen.z

Now we know.

**Pull request merged**